File: patterns/java/src/refactoring_guru/patterns/observer/example/LogOpenListener.java
package refactoring_guru.patterns.observer.example;

import java.io.File;
import java.io.FileWriter;
import java.io.IOException;

public class LogOpenListener implements EventListener {
    private File log;

    public LogOpenListener(String fileName) {
        this.log = new File(fileName);
    }

    public void update(File file) {
        try{
            FileWriter writer = new FileWriter(log);
            writer.write("Opened: " + file.getCanonicalPath());
            writer.flush();
        } catch (IOException ex) {
            ex.printStackTrace();
        }
    }
}

File: patterns/cs/src/RefactoringGuru/Observer/Example/Listeners/LogOpenListener.cs
using System;
using System.IO;

namespace RefactoringGuru.Observer.Example.Listeners
{
    class LogOpenListener : EventListener
    {
        private FileInfo log;

        public LogOpenListener(String fileName)
        {
            this.log = new FileInfo(fileName);
        }
        public void Update(string eventType, FileInfo file)
        {
            Console.WriteLine("Save to log " + log + ": Someone has performed " + eventType + " operation with the following file: " + file.Name);
        }
    }
}
